motleycrew/agents/parent.py

```
@@ -111,6 +111,13 @@ def compose_prompt(
         Returns:
             The composed prompt.
         """
+        # If input is already a list of messages, return them directly
+        if isinstance(input, list) and all(isinstance(m, BaseMessage) for m in input):
+            if as_messages:
+                return input
+            # Convert messages to string representation (fallback)
+            return "\n\n".join([m.content if hasattr(m, 'content') and isinstance(m.content, str) else str(m.content) for m in input])
+        
```

We should at least raise an exception if self.prompt is not None in this case, because now it'll just be silently ignored

---

Changed to prepend self.prompt before the list of messages if input is a list of messages

---

where?

---

see https://github.com/MotleyAI/motleycrew/pull/111/files#r2306989609

---

?